portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -83,9 +76,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       response.sendRedirect("/index.html");
   }
 
-  private String convertToJsonUsingGson(ArrayList<Comment> comments) {
+  private String convertToJsonUsingGson(ArrayList<Comment> comments, int numComments) {
     Gson gson = new Gson();
-    String json = gson.toJson(comments);
+
+    JsonObject obj = new JsonObject();
+    String jsonComments = gson.toJson(comments);
+    obj.addProperty("comments", jsonComments);
```

Comment:
On my comment below to avoid having to JSON.parse twice, can you just set this to comments here instead of setting it to the string version of comments?

Reply:
fixed, I basically ended up having to convert comments to be Json-compatible